dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -1615,6 +1645,9 @@ private void killAllTasks() {
             }
             TaskInstance taskInstance = processService.findTaskInstanceById(taskInstanceId);
             if (taskInstance == null || taskInstance.getState().typeIsFinished()) {
+                if (readyToSubmitTaskQueue.size() > 0) {
```

Comment:
Why to clear submit task queue when some taskInstance is null or finish?

Reply:
When the task instance fails and the kill command is received in the task retry interval. We need to clear the readyToSubmitTaskQueue to ensure that the subsequent checks are normal.